src/utils/credentials.ts

```diff
@@ -1,50 +1,66 @@
 import { Credential, DB } from '../types';
 import { readFile, writeFile } from 'fs/promises';
+import { deCrypt, enCript } from './crypto';
 
-export async function readCredentials(): Promise<Credential[]> {
+export async function readCredentials(key: string): Promise<Credential[]> {
   const response = await readFile('src/db.json', 'utf-8');
   const db: DB = JSON.parse(response);
-  const credentials = db.credentials;
-  return credentials;
+  const credentials: Credential[] = db.credentials;
+  const deCryptCredentials = credentials.map((credential) =>
+    deCrypt(credential, key)
+  );
+  return deCryptCredentials;
 }
-export async function getCredential(service: string): Promise<Credential> {
-  const credentials = await readCredentials();
+export async function getCredential(
+  service: string,
+  key: string
+): Promise<Credential> {
+  const credentials = await readCredentials(key);
   const credential = credentials.find(
     (credential) => credential.service.toLowerCase() === service.toLowerCase()
   );
+
   if (!credential) {
     throw new Error(`ERROR, NO such Service WuuuUHahaHA !!!: ${service}`);
   }
+  //   const deCryptCredential = deCrypt(credential, key);
   return credential;
 }
-export async function deleteCredential(service: string): Promise<void> {
-  const credentials = await readCredentials();
+export async function deleteCredential(
+  service: string,
+  key: string
+): Promise<void> {
+  const credentials = await readCredentials(key);
   const filteredCredentials = credentials.filter(
     (credential) => credential.service.toLowerCase() !== service.toLowerCase()
   );
   const db: DB = { credentials: filteredCredentials };
   overWriteDB(db);
 }
-
-export async function addCredential(credential: Credential): Promise<void> {
-  const response = await readFile('src/db.json', 'utf-8');
-  const db: DB = JSON.parse(response);
-  db.credentials = [...db.credentials, credential];
-  overWriteDB(db);
-}
 async function overWriteDB(db: DB) {
   const dbString = JSON.stringify(db, null, 2);
   await writeFile('src/db.json', dbString);
 }
-
 export async function updateCredential(
   service: string,
-  credential: Credential
+  credential: Credential,
+  key: string
 ): Promise<void> {
-  const credentials = await readCredentials();
+  const credentials = await readCredentials(key);
```

Comment:
Be careful as this will decrypt your whole DB and you are later inserting everything in credentials into the new DB therefore saving the plain text passwords